blueocean-core-js/src/js/services/ActivityService.js

```diff
@@ -72,7 +72,7 @@ export class ActivityService extends BunkerService {
                 // Should really have dedupe on methods like these, but for now
                 // just clone data so that we dont modify other instances.
                 const run = utils.clone(data);
-                return this.setItem(run);
+                return this.setItemWithKey(href, run);
```

Comment:
why is this needed? shouldnt the runh ave the self href in it?

Reply:
yes I did want to know why this is here. 

Maybe @cliffmeyers 

@tinexw does this fix some latent bug you saw?

Reply:
It does have the self href but it points to the actual run url, not the latest url.
E.g. in the example from the JIRA ticket ```https://ci.blueocean.io/blue/rest/organizations/jenkins/pipelines/ATH-Jenkinsfile/branches/master/latestRun/```contains 

```
_links: {
    self: {
        _class: "io.jenkins.blueocean.rest.hal.Link",
        href: "/blue/rest/organizations/jenkins/pipelines/ATH-Jenkinsfile/branches/master/runs/2433/"
},
```

The latest URL is used to as key to fetch and get the latest run from the activity service (in ``RunDetails.jsx``) but it is stored internally in the bunker service with the run id URL. Thus, it is not found from the run details.

Another option would be to change the backend response - I briefly looked into it but it seemed more complicated and kind of wrong anyway.

Reply:
We shouldn't do this, why not just redirect to the actual run url and avoid this bug altogether?

Reply:
I'd much rather not introduce known bugs.

Reply:
What known bug do you mean? 
I thought about just doing a redirect too but then wouldn't you show the redirected URL in the browser? How would the refresh work?

Reply:
known bug that the bunker has a `latestRun` cached and won't re-fetch it?

Reply:
Ok, didn't realize this. I only tested manually that the newest run is displayed when I refreshed the latest run page and started a new run in between.

Reply:
@kzantow Maybe just to clarify: What I described above is a bug that would be there **without** the changes to the ActivityService and BunkerService. With these changes the bug does not exist. I just tried to explain why I made those changes i.e. to prevent this bug from happening.